### PR TITLE
OpenShift 0.5 compatibility

### PR DESCRIPTION
**Pull request by mmilata**

### CODE REVIEW

File: osbs/core.py [hunk 1]
```diff
@@ -175,16 +171,15 @@ def logs(self, build_id, follow=False, namespace=DEFAULT_NAMESPACE):
...
-            # 0.4.1
-            # FIXME: remove this once 0.4.3 is deployed everywhere
-            buildlogs_url = self._build_k8s_url(
-                "containerLogs/default/build-%s/custom-build?follow=%d" % (
-                    build_id, 1 if follow else 0))
+            # 0.4.3
+            # FIXME: remove this once 0.5.? is deployed everywhere
+            buildlogs_url = self._build_url("proxy/buildLogs/%s/" % build_id,
+                    follow=(1 if follow else 0), namespace=namespace)
             response.close_multi()
```
Comment (mmilata) [inline osbs/core.py]: I'm not sure what this does and if it should stay here even if we are GETting from the same host (which was previously not the case).

**Comment by mmilata** [comment]
(Will fix pep8 issues, do not merge yet.)

File: osbs/core.py [hunk 2]
```diff
@@ -175,16 +171,17 @@ def logs(self, build_id, follow=False, namespace=DEFAULT_NAMESPACE):
...
-            # FIXME: remove this once 0.4.3 is deployed everywhere
-            buildlogs_url = self._build_k8s_url(
-                "containerLogs/default/build-%s/custom-build?follow=%d" % (
-                    build_id, 1 if follow else 0))
+            # 0.4.3
+            # FIXME: remove this once 0.5.? is deployed everywhere
+            buildlogs_url = self._build_url("proxy/buildLogs/%s/" % build_id,
+                                            follow=(1 if follow else 0),
+                                            namespace=namespace)
             response.close_multi()
```
Comment (mmilata) [inline osbs/core.py]: I'm not sure what this does and if it should stay here even if we are GETting from the same host (which was previously not the case).

Comment (TomasTomecek) [inline osbs/core.py]: The problem was, that curl handle stayed open on the multi request and subsequent request raised an exception. I guess we could implement it as context manager.

File: osbs/core.py [hunk 3]
```diff
@@ -59,15 +58,12 @@ def __init__(self, openshift_api_url, openshift_oauth_url, kubelet_base, verbose
...
-        if namespace:
-            url += "?namespace=%s" % namespace
-            return urlparse.urljoin(self.os_api_url, url)
-        else:
-            return urlparse.urljoin(self.os_api_url, url)
-
-    def _build_k8s_url(self, url):
-        return urlparse.urljoin(self.kubelet_base, url)
+    def _build_url(self, url, **query):
+        query = [k + '=' + str(v) for k, v in query.items()]
```
Comment (TomasTomecek) [inline osbs/core.py]: Am checking if urlparse doesn't have method for this but looks like it doesn't =/

File: osbs/core.py (hunk 2 above)
Comment (mmilata) [inline osbs/core.py]: So that means we should keep the `response.close_multi()` line for now, right?

File: osbs/core.py (hunk 3 above)
Comment (mmilata) [inline osbs/core.py]: Actually there's `urllib.urlencode` and `urllib.parse.urlencode` (py3). I'll update the PR.

File: osbs/core.py (hunk 2 above)
Comment (TomasTomecek) [inline osbs/core.py]: yes, definitely; it could be great to add comment there so anyone doesn't remove it by accident

**Comment by TomasTomecek** [comment]
Merging and am going to give it a try. Thanks Martin!

**Comment by twaugh** [comment]
Would be great to get add `tests/mock_jsons/0.5.x` once it's working.

**Comment by mmilata** [comment]
@twaugh good idea! I'd like to add them once 0.5.2 is out and tested against osbs (0.5.1 has a bug that prevents it from working with osbs using the v1beta1 api).

**Comment by TomasTomecek** [comment]
TBH, that shell script is really bad and I feel bad. I think we should rewrite it to python (and ideally use `osbs.http`)

**Pull request merged**